spec/forms/search_developer_form_spec.rb

```diff
@@ -3,16 +3,92 @@
 
 RSpec.describe SearchDeveloperForm do
   describe '#search' do
+    before do
+      @developer1 = create(:developer, email: 'abc2321@gmail.com')
+      @developer2 = create(:developer, email: 'abc2121@gmail.com')
+      @developer3 = create(:developer, email: 'abc2341@gmail.com')
+    end
     context 'with no params' do
+      it 'returns no developers' do
+        form = SearchDeveloperForm.new({})
+        expect(form.search.size).to eq(0)
+      end
+    end
+    context 'with params' do
       before do
-        create(:developer, email: 'abc@test.com')
-        create(:developer, email: 'abd@test.com')
+        @language1 = create(:language, code: 'vn')
+        @language2 = create(:language, code: 'en')
+        @language3 = create(:language, code: 'jp')
+
+        @programming_language1 = create(:programming_language, name: 'ruby')
+        @programming_language2 = create(:programming_language, name: 'js')
+        @programming_language3 = create(:programming_language, name: 'c++')
+
+        @developer_language1 = create(:developer_language, developer_id: @developer1.id, language_id: @language1.id)
+        @developer_language2 = create(:developer_language, developer_id: @developer2.id, language_id: @language2.id)
+        @developer_language3 = create(:developer_language, developer_id: @developer3.id, language_id: @language3.id)
+        @developer_language4 = create(:developer_language, developer_id: @developer1.id, language_id: @language3.id)
```

Comment:
The four variables seem not referred from anywhere.
So you don't need to use variables.
You can just do 
`create(:developer_language, developer_id: @developer1.id, language_id: @language3.id)`